### PR TITLE
Tests: typecheck SafeDITool-generated code against its inputs

### PR DESCRIPTION
**Pull request by dfed**
## Summary
- Add `SafeDICompileVerifier` (and `SafeDIBuildArtifactLocator`) that invokes `swiftc -typecheck` on the fixture sources plus generated output. Wire it into `executeSafeDIToolTest` so every test that round-trips through the generator also verifies the output compiles against the built `SafeDI` module and `SafeDIMacros` plugin.
- Two-pass verification: inputs alone first (so fixture-Swift errors surface cleanly), then inputs plus generated output. `SAFEDI_SKIP_COMPILE_CHECK=1` is an escape hatch for fast local iteration.
- Mark tests that cannot be typechecked in isolation — cross-module placeholders, Linux-specific platform stubs, codegen with preexisting bugs — with `skipCompileVerification: true` and a comment explaining why. Leaves a paper trail for follow-up work.
- Fix ~10 fixture-level issues the verifier surfaced immediately (invalid class stand-ins, missing `Instantiable` conformance, platform-only types standing in as protocols, etc.).

## Self-review pass (commit `e5b91c1`)
- Corrected the `skipCompileVerification` reason comment on two internal-Child tests: the verifier compiles everything in one synthetic module, so the real failure is Swift's same-module rule about `public init` signatures referencing `internal` types — not cross-module inaccessibility.

## Why

Before this PR, generator regressions that produced malformed-but-plausible Swift slipped through the test suite unless they tripped an integration test. The mock output strings were compared character-for-character, but nothing confirmed that the strings were valid Swift against the rest of the module.

## Test plan
- [x] `swift test --traits sourceBuild` — 841 tests pass
- [x] `./CLI/lint.sh`
- [x] Codecov patch coverage — [check CI]

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by dfed** [comment]
@codex review

File: Tests/SafeDIToolTests/Helpers/SafeDICompileVerifier.swift [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+	skippingTypes: Set<String> = [],
+) throws -> String {
+	// Matches:
+	//   @Instantiable[(optional arg list)]
+	//   [attributes/whitespace/newlines]
+	//   [access modifier] [final] (class|struct|actor|extension) TypeName[: conformances]? {
+	//
+	// Captures the header up through the opening `{` so we can inspect the
+	// conformance clause without swallowing the body.
+	let pattern = #"(@Instantiable(?:\s*\([^)]*\))?)(\s+(?:[^{]*?))((?:public|internal|open|package|fileprivate|private)?\s*(?:final\s+)?(?:class|struct|actor|extension)\s+(\w+)(\s*:\s*[^{]*)?\s*\{)"#
```
[the remark this answers is not in the document]
Comment (dfed) [reply Tests/SafeDIToolTests/Helpers/SafeDICompileVerifier.swift]: The regex-based conformance injection was removed; fixtures now declare `Instantiable` conformance directly and the verifier just runs `swiftc -typecheck` on them. Resolving.

File: Tests/SafeDIToolTests/Helpers/SafeDICompileVerifier.swift [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+		guard !alreadyConforms else { return }
+		// Don't inject when conformance has already been declared on the same
+		// type in another fixture file — Swift would treat it as redundant.
+		guard !skippingTypes.contains(typeName) else { return }
+
+		var rewrittenHeader = header
+		if let conformanceText {
+			// Append ", Instantiable" to the existing conformance list.
+			let trimmed = conformanceText.trimmingCharacters(in: .whitespaces)
+			let replacement = trimmed + ", Instantiable "
```
[the remark this answers is not in the document]
Comment (dfed) [reply Tests/SafeDIToolTests/Helpers/SafeDICompileVerifier.swift]: Same as above — the regex-based injection was removed. Fixtures declare `Instantiable` conformance directly now. Resolving.

File: Tests/SafeDIToolTests/Helpers/SafeDICompileVerifier.swift [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+	let stdoutPipe = Pipe()
+	let stderrPipe = Pipe()
+	process.standardOutput = stdoutPipe
+	process.standardError = stderrPipe
+
+	try process.run()
+	process.waitUntilExit()
+
+	let stdoutData = try stdoutPipe.fileHandleForReading.readToEnd() ?? Data()
+	let stderrData = try stderrPipe.fileHandleForReading.readToEnd() ?? Data()
```
[the remark this answers is not in the document]
Comment (dfed) [reply Tests/SafeDIToolTests/Helpers/SafeDICompileVerifier.swift]: Fixed. `runSwiftTypecheck` now attaches readability handlers to stdout/stderr that accumulate into a thread-safe `LockedData` while the process runs, so a chatty `swiftc` failure can't deadlock by filling the OS pipe buffer.

File: Tests/SafeDIToolTests/Helpers/SafeDICompileVerifier.swift [hunk 4]
```diff
@@ -0,0 +1,371 @@
...
+	skippingTypes: Set<String> = [],
+) throws -> String {
+	// Matches:
+	//   @Instantiable[(optional arg list)]
+	//   [attributes/whitespace/newlines]
+	//   [access modifier] [final] (class|struct|actor|extension) TypeName[: conformances]? {
+	//
+	// Captures the header up through the opening `{` so we can inspect the
+	// conformance clause without swallowing the body.
+	let pattern = #"(@Instantiable(?:\s*\([^)]*\))?)(\s+(?:[^{]*?))((?:public|internal|open|package|fileprivate|private)?\s*(?:final\s+)?(?:class|struct|actor|extension)\s+(\w+)(\s*:\s*[^{]*)?\s*\{)"#
```
[the remark this answers is not in the document]
Comment (dfed) [reply Tests/SafeDIToolTests/Helpers/SafeDICompileVerifier.swift]: The regex-based injection was removed; fixtures declare `Instantiable` conformance directly. Resolving.

File: Tests/SafeDIToolTests/Helpers/SafeDICompileVerifier.swift [hunk 5]
```diff
@@ -0,0 +1,386 @@
...
+	}
+
+	// First pass: typecheck inputs alone. If the inputs can't compile on their
+	// own (broken-on-purpose fixtures, cross-module references, etc.), skip
+	// the output verification entirely.
+	let inputsOnlyResult = try runSwiftTypecheck(
+		sources: inputCompileFiles,
+		artifacts: artifacts,
+	)
+	if inputsOnlyResult.exitCode != 0 { return }
```
[the remark this answers is not in the document]
Comment (dfed) [reply Tests/SafeDIToolTests/Helpers/SafeDICompileVerifier.swift]: `Issue.record` is called (not silently returning), so CI fails when the inputs-only pass fails — the failure surfaces as a test issue. Agreed the current message blames the fixture, which can be misleading when the root cause is infrastructure (plugin load, stale `.build` artifacts). I'll broaden the diagnostic copy.

File: Tests/SafeDIToolTests/Helpers/SafeDICompileVerifier.swift [hunk 6]
```diff
@@ -0,0 +1,386 @@
...
+	skippingTypes: Set<String> = [],
+) throws -> String {
+	// Matches:
+	//   @Instantiable[(optional arg list)]
+	//   [attributes/whitespace/newlines]
+	//   [access modifier] [final] (class|struct|actor|extension) TypeName[: conformances]? {
+	//
+	// Captures the header up through the opening `{` so we can inspect the
+	// conformance clause without swallowing the body.
+	let pattern = #"(@Instantiable(?:\s*\([^)]*\))?)(\s+(?:[^{]*?))((?:public|internal|open|package|fileprivate|private)?\s*(?:final\s+)?(?:class|struct|actor|extension)\s+(\w+)(\s*:\s*[^{]*)?\s*\{)"#
```
[the remark this answers is not in the document]
Comment (dfed) [reply Tests/SafeDIToolTests/Helpers/SafeDICompileVerifier.swift]: The regex-based injection was removed. Resolving.

File: Tests/SafeDIToolTests/Helpers/SafeDICompileVerifier.swift [hunk 7]
```diff
@@ -0,0 +1,584 @@
...
+}
+
+/// Strips a module qualifier and any generic parameter list from a captured
+/// type-declaration name, leaving the bare identifier (`MyModule.Container<T>`
+/// → `Container`). Used as the canonical key for the
+/// `typesAlreadyDeclaringConformance` skip set.
+private func extractBaseTypeName(from qualified: String) -> String {
+	let withoutGenerics = qualified.split(separator: "<", maxSplits: 1, omittingEmptySubsequences: false)[0]
+	if let lastDot = withoutGenerics.lastIndex(of: ".") {
+		return String(withoutGenerics[withoutGenerics.index(after: lastDot)...])
```
[the remark this answers is not in the document]
Comment (dfed) [reply Tests/SafeDIToolTests/Helpers/SafeDICompileVerifier.swift]: The regex-based injection was removed. Resolving.

File: Sources/SafeDICore/Generators/ScopeGenerator.swift [hunk 8]
```diff
@@ -1850,6 +1852,11 @@ actor ScopeGenerator: CustomStringConvertible, Sendable {
 				} else {
 					"\(nodePath).\(argument.label)"
 				}
+			} else if argument.label == "_", let defaultExpression = argument.defaultValueExpression {
+				// Underscore-labeled default-valued parameters are not bubbled as
+				// overrides (they have no label to surface on the config struct),
+				// so inline the default expression at the call site.
+				defaultExpression
```
[the remark this answers is not in the document]
Comment (dfed) [reply Sources/SafeDICore/Generators/ScopeGenerator.swift]: Good catch — inlining an underscore-labeled default is fine for literal defaults (what the current tests use) but brittle for `Self.<member>`, private members, or file-scoped symbols. Tracking as a follow-up; the right fix is to omit trailing unlabeled-default arguments from the generated call entirely so Swift's default-argument thunk fires in the callee's declaration context.

File: Tests/SafeDIToolTests/Helpers/SafeDICompileVerifier.swift [hunk 9]
```diff
@@ -0,0 +1,361 @@
...
+	sourceLocation: SourceLocation = #_sourceLocation,
+) throws {
+	guard ProcessInfo.processInfo.environment["SAFEDI_SKIP_COMPILE_CHECK"] == nil else { return }
+
+	// Skip fixtures that depend on platform-specific frameworks (UIKit, SwiftUI,
+	// AppKit, WatchKit) — the SafeDI artifacts are built for the host macOS
+	// target, and retargeting the verifier for iOS/watchOS/etc. would require
+	// re-compiling the package for every OS in CI. The production user's build
+	// (which has the right SDK) will still catch framework-level issues.
+	let platformFrameworks = ["UIKit", "SwiftUI", "AppKit", "WatchKit", "Cocoa"]
```
[the remark this answers is not in the document]
Comment (dfed) [reply Tests/SafeDIToolTests/Helpers/SafeDICompileVerifier.swift]: Fixed. The SwiftUI-importing fixtures in `SafeDIToolCodeGenerationTests` were converted to declare local `View` / `AnyView` / `EmptyView` stubs, so the verifier no longer needs to skip based on `canImport`.

File: Tests/SafeDIToolTests/Helpers/SafeDIToolTestExecution.swift [hunk 10]
```diff
@@ -179,6 +180,26 @@ func executeSafeDIToolTest(
...
+		if let generatedFiles, !generatedFiles.isEmpty {
+			// Skip compile verification when the test simulates cross-module
+			// scenarios (types declared in the main fixture reference types
+			// that only exist in a sibling module) or when the caller has
+			// explicitly opted out. The checker needs every referenced type
+			// to be defined in the compilation unit; cross-module fixtures
+			// violate that contract on purpose.
+			let shouldVerify = !skipCompileVerification
+				&& dependentModuleInfoPaths.isEmpty
+				&& additionalImportedModules.isEmpty
```
[the remark this answers is not in the document]
Comment (dfed) [reply Tests/SafeDIToolTests/Helpers/SafeDIToolTestExecution.swift]: Fixed in 37242df. The blanket skip on `additionalImportedModules` was dropped; the two existing tests that pass the placeholder `"Test"` module now opt out explicitly via `skipCompileVerification: true`, so future tests passing real host-available modules (Foundation/Combine/...) get verified.

File: Tests/SafeDIToolTests/Helpers/SafeDICompileVerifier.swift [hunk 11]
```diff
@@ -0,0 +1,326 @@
...
+	#else
+		process.executableURL = URL(fileURLWithPath: "/usr/bin/xcrun")
+		var arguments = ["swiftc"]
+	#endif
+	arguments.append(contentsOf: [
+		"-typecheck",
+		"-module-name", "SafeDICodegenVerification",
+		"-I", artifacts.swiftModuleSearchPath.path,
+		"-load-plugin-executable", "\(artifacts.safeDIMacrosToolPath.path)#SafeDIMacros",
+		"-DDEBUG",
```
[the remark this answers is not in the document]
Comment (dfed) [reply Tests/SafeDIToolTests/Helpers/SafeDICompileVerifier.swift:268]: Valid, deferred to a follow-up. Adding `-swift-version 6` locally surfaces 6 real issues — 4 `mock_sendable*` tests hit non-Sendable closure captures inside `@Sendable` local functions (genuine generator bugs), plus 2 fixture-level strict-concurrency issues (`UserDefaults.standard`, `Sendable` class with non-Sendable stored properties). That's a meaningful scope expansion into Swift-6-strict-concurrency work; want to ship the verifier first and tackle Swift 6 compliance separately.

File: Sources/SafeDICore/Generators/ScopeGenerator.swift [hunk 12]
```diff
@@ -540,6 +540,8 @@ actor ScopeGenerator: CustomStringConvertible, Sendable {
 				}
 				let initializer = if !hasGeneratedContent {
 					existentialWrappedReturn
+				} else if erasedToConcreteExistential {
+					"\(property.typeDescription.asSource)(\(functionName)())"
```
[the remark this answers is not in the document]
Comment (dfed) [reply Sources/SafeDICore/Generators/ScopeGenerator.swift:544]: I think this is a false positive. `erasedToConcreteExistential` is documented as being for concrete wrapper types like `AnyView` that expose a no-label `init(_:)` — see `Instantiated.swift` / `Received.swift` API docs. Both the existing `!hasGeneratedContent` branch (line 537) and the new `hasGeneratedContent` branch emit `PropertyType(…)`; the new branch just mirrors the older pattern. Bare protocols without a callable init fail both branches, but the API contract requires a callable wrapper, so this isn't a regression for valid inputs.

**Comment by dfed** [comment]
@codex review

File: Sources/SafeDICore/Generators/ScopeGenerator.swift [hunk 13]
```diff
@@ -1488,18 +1579,17 @@ actor ScopeGenerator: CustomStringConvertible, Sendable {
 				initParameters.append("\(innerIndent)\(standardIndent)\(label): \(child.configurationTypeName) = .init()")
 				storedProperties.append("\(innerIndent)let \(label): \(child.configurationTypeName)")
 			} else {
-				let childSendable = child.requiresSendable ? "@Sendable " : ""
-				initParameters.append("\(innerIndent)\(standardIndent)\(label): (\(childSendable)\(child.builderClosureType))? = nil")
-				storedProperties.append("\(innerIndent)let \(label): (\(childSendable)\(child.builderClosureType))?")
+				initParameters.append("\(innerIndent)\(standardIndent)\(label): (\(sendableAnnotation)\(child.builderClosureType))? = nil")
+				storedProperties.append("\(innerIndent)let \(label): (\(sendableAnnotation)\(child.builderClosureType))?")
```
[the remark this answers is not in the document]
Comment (dfed) [reply Sources/SafeDICore/Generators/ScopeGenerator.swift]: Fixed in 40e41fd — leaf child fields now use \`requiresSendable || child.requiresSendable\` so a sendably-reached child keeps \`@Sendable\` on its slot regardless of the parent. Regression test \`mock_leafSendableInstantiatorInsideNonSendableParentMarksChildFieldSendable\` covers the exact scenario (non-sendable parent containing a leaf \`SendableInstantiator<Child>\`).

[the remark this answers is not in the document]
Comment (dfed) [reply Sources/SafeDICore/Generators/ScopeGenerator.swift]: Fixed in 40e41fd — leaf child fields now use \`requiresSendable || child.requiresSendable\` so a sendably-reached child keeps \`@Sendable\` on its slot regardless of the parent. Regression test \`mock_leafSendableInstantiatorInsideNonSendableParentMarksChildFieldSendable\` covers the exact scenario (non-sendable parent containing a leaf \`SendableInstantiator<Child>\`).

File: Sources/SafeDICore/Generators/ScopeGenerator.swift (hunk 8 above)
[the remark this answers is not in the document]
Comment (dfed) [reply Sources/SafeDICore/Generators/ScopeGenerator.swift]: Addressed: the generator no longer inlines default expressions at the call site. `callSiteArguments` filters out unlabeled defaulted non-dependency params, and `defaultBuilderExpression` emits `{ Type(label: $0) }` when needed so Swift's default-argument thunk fires in the callee's declaration context (preserves `Self`-binding and unqualified name lookup inside the default). Note that a proper long-term fix would be for the `@Instantiable` macro to reject `Self.*` (and other callee-scope-only references) in default expressions, tracked as follow-up.

File: Sources/SafeDICore/Generators/ScopeGenerator.swift [hunk 14]
```diff
@@ -1073,13 +1075,28 @@ actor ScopeGenerator: CustomStringConvertible, Sendable {
 			"\(concreteType.asSource).\(Self.configurationStructName)"
 		}
 
+		/// Construction arguments surfaced at the mock call site. Unlabeled
+		/// parameters with defaults are omitted so Swift's default-argument thunk
+		/// fires in the callee's declaration context, preserving resolution of
+		/// `Self.*`, private members, and file-scoped symbols in the default
+		/// expression that would otherwise fail when inlined at the caller.
+		var callSiteArguments: [Initializer.Argument] {
+			constructionArguments.filter { !($0.label == "_" && $0.hasDefaultValue) }
```
[the remark this answers is not in the document]
Comment (dfed) [reply Sources/SafeDICore/Generators/ScopeGenerator.swift]: Addressed: the `callSiteArguments` filter preserves unlabeled args whose inner label matches one of the type's dependencies (`!dependencyInnerLabels.contains(argument.innerLabel)` carve-out). Regression test `mock_threadsResolvedBinding_whenDependencyParameterIsUnlabeledWithDefault` pins the behavior.

File: Sources/SafeDICore/Generators/DependencyTreeGenerator.swift [hunk 15]
```diff
@@ -110,6 +110,15 @@ public actor DependencyTreeGenerator {
...
 		)
 
+		// Reject initializer shapes that SafeDI's mock override surface cannot
+		// express. A non-dependency `_`-labeled defaulted parameter is normally
+		// elided so Swift's default-argument thunk fires in the callee; but if
+		// any subsequent parameter is also `_`-labeled, Swift still requires the
+		// positional slot to be passed, so eliding would generate an uncallable
+		// signature. Exposing it on `SafeDIMockConfiguration` isn't an option
+		// either — there's no external label to use as a struct field name.
+		try validateMockableInitializerShapes(typeDescriptionToScopeMap: typeDescriptionToScopeMap)
```
[the remark this answers is not in the document]
Comment (dfed) [reply Sources/SafeDICore/Generators/DependencyTreeGenerator.swift]: Addressed by moving the check from the plugin to the macro. Per CLAUDE.md's validation-boundary guidance (macro for local-context checks with fix-its; plugin for cross-type), the init shape check is intrinsically per-type, so `InstantiableMacro` now emits the diagnostic via `validateUnlabeledDefaultShape` on each candidate initializer/mock method. Moving it to the macro fixes (a) scope (only fires for decorated types, no abort-for-unrelated-roots) and (b) the predicate (now allows subsequent `_`-labeled parameters that also have defaults, since those are elided together as a trailing suffix). Macro tests: `producesDiagnostic_whenUnlabeledDefaultPrecedesUnlabeledRequiredParameterOnInit` and `producesNoDiagnostic_whenTrailingUnlabeledParametersAllHaveDefaults`. Plugin-side validation + error case removed.

File: Sources/SafeDICore/Generators/DependencyTreeGenerator.swift [hunk 16]
```diff
@@ -958,6 +993,41 @@ public actor DependencyTreeGenerator {
...
+			let instantiable = scope.instantiable
+			let dependencyLabels = Set(instantiable.dependencies.map(\.property.label))
+			let initializer = instantiable.mockInitializer ?? instantiable.initializer
+			guard let arguments = initializer?.arguments else { continue }
+			for (index, argument) in arguments.enumerated() {
+				guard argument.label == "_",
+				      argument.hasDefaultValue,
+				      !dependencyLabels.contains(argument.innerLabel)
+				else { continue }
+				if let following = arguments[(index + 1)...].first(where: { $0.label == "_" }) {
```
[the remark this answers is not in the document]
Comment (dfed) [reply Sources/SafeDICore/Generators/DependencyTreeGenerator.swift]: Addressed by moving the check from the plugin to the macro. Per CLAUDE.md's validation-boundary guidance (macro for local-context checks with fix-its; plugin for cross-type), the init shape check is intrinsically per-type, so `InstantiableMacro` now emits the diagnostic via `validateUnlabeledDefaultShape` on each candidate initializer/mock method. Moving it to the macro fixes (a) scope (only fires for decorated types, no abort-for-unrelated-roots) and (b) the predicate (now allows subsequent `_`-labeled parameters that also have defaults, since those are elided together as a trailing suffix). Macro tests: `producesDiagnostic_whenUnlabeledDefaultPrecedesUnlabeledRequiredParameterOnInit` and `producesNoDiagnostic_whenTrailingUnlabeledParametersAllHaveDefaults`. Plugin-side validation + error case removed.

File: Sources/SafeDICore/Generators/DependencyTreeGenerator.swift (hunk 15 above)
[the remark this answers is not in the document]
Comment (dfed) [reply Sources/SafeDICore/Generators/DependencyTreeGenerator.swift]: Addressed by moving the check from the plugin to the macro. Per CLAUDE.md's validation-boundary guidance (macro for local-context checks with fix-its; plugin for cross-type), the init shape check is intrinsically per-type, so `InstantiableMacro` now emits the diagnostic via `validateUnlabeledDefaultShape` on each candidate initializer/mock/instantiate method. Moving it to the macro fixes (a) scope — only fires for decorated types, no abort-for-unrelated-roots — and (b) the predicate: now allows subsequent `_`-labeled parameters that also have defaults, since those are elided together as a trailing suffix. Macro tests: `producesDiagnostic_whenUnlabeledDefaultPrecedesUnlabeledRequiredParameterOnInit` and `producesNoDiagnostic_whenTrailingUnlabeledParametersAllHaveDefaults`. Plugin-side validation + error case removed.

File: Sources/SafeDICore/Generators/DependencyTreeGenerator.swift (hunk 16 above)
[the remark this answers is not in the document]
Comment (dfed) [reply Sources/SafeDICore/Generators/DependencyTreeGenerator.swift]: Addressed by moving the check from the plugin to the macro. Per CLAUDE.md's validation-boundary guidance (macro for local-context checks with fix-its; plugin for cross-type), the init shape check is intrinsically per-type, so `InstantiableMacro` now emits the diagnostic via `validateUnlabeledDefaultShape` on each candidate initializer/mock/instantiate method. Moving it to the macro fixes (a) scope — only fires for decorated types, no abort-for-unrelated-roots — and (b) the predicate: now allows subsequent `_`-labeled parameters that also have defaults, since those are elided together as a trailing suffix. Macro tests: `producesDiagnostic_whenUnlabeledDefaultPrecedesUnlabeledRequiredParameterOnInit` and `producesNoDiagnostic_whenTrailingUnlabeledParametersAllHaveDefaults`. Plugin-side validation + error case removed.

File: Sources/SafeDIMacros/Macros/InstantiableMacro.swift [hunk 17]
```diff
@@ -1028,6 +1062,22 @@ public struct InstantiableMacro: MemberMacro {
...
+			// methods. Matches the concrete-type validation above.
+			for member in extensionDeclaration.memberBlock.members {
+				guard let function = FunctionDeclSyntax(member.decl),
+				      function.name.text == InstantiableVisitor.instantiateMethodName
+				else { continue }
+				let initializer = Initializer(function)
+				Self.validateUnlabeledDefaultShape(
+					functionSyntax: function.signature,
+					arguments: initializer.arguments,
+					dependencies: visitor.dependencies,
```
[the remark this answers is not in the document]
Comment (dfed) [reply Sources/SafeDIMacros/Macros/InstantiableMacro.swift]: Fixed in 739ea35. An extension's `instantiate()` method can't trip this diagnostic — every argument is a dependency by construction — so the extension-side `instantiate` walk is removed entirely. The remaining extension-side callers (mock methods) now pass `extensionDependencies` (the flattened `visitor.instantiables.flatMap(\.dependencies)`) instead of the empty `visitor.dependencies`. Bonus: the diagnostic now carries a fix-it that promotes `_ x: T = …` to `x: T = …`.

File: Sources/SafeDICore/Generators/ScopeGenerator.swift [hunk 18]
```diff
@@ -1853,24 +1953,24 @@ actor ScopeGenerator: CustomStringConvertible, Sendable {
...
-					"\(sendableExtractionPrefix)__\(relativePath)_\(argument.label)"
-				} else {
-					"\(nodePath).\(argument.label)"
-				}
+			} else if let sendableExtractionPrefix {
+				"\(sendableExtractionPrefix)__\(relativePath)_\(argument.label)"
 			} else {
-				// Unknown argument — use the label as a local variable reference.
-				argument.innerLabel
+				"\(nodePath).\(argument.label)"
```
[the remark this answers is not in the document]
Comment (dfed) [reply Sources/SafeDICore/Generators/ScopeGenerator.swift:1976]: Investigated — this is a false positive. Every non-dependency argument surviving `callSiteArguments` is a labeled default, so the `\(nodePath).\(argument.label)` fallback always resolves to a real `SafeDIMockConfiguration` field.

Required non-default non-dependency arguments are rejected at macro-expansion time before SafeDITool ever sees them:

- **Regular `init`**: `InstantiableMacro.swift:246` calls `initializer.validate(fulfilling:)`, which runs `createDependencyAndArgumentBinding` (`Initializer.swift:213`) and throws `unexpectedArgument` for any arg that is neither a dependency nor has a default.
- **`customMock` methods**: `InstantiableMacro.swift:596` / `:923` emit `mockMethodNonDependencyMissingDefaultValue` (with a fix-it inserting `<#default#>`), and `:681` / `:1028` additionally run `validate(fulfilling:)` on the mock method.

No code change needed.

File: Tests/SafeDIToolTests/Helpers/SafeDICompileVerifier.swift [hunk 19]
```diff
@@ -0,0 +1,332 @@
...
+	guard let artifacts = SafeDIBuildArtifactLocator.locate() else {
+		Issue.record(
+			"""
+			Could not locate SafeDI build artifacts for compilation verification. \
+			Ensure the package is built before running tests, or set \
+			SAFEDI_SKIP_COMPILE_CHECK=1 to skip.
+			""",
+			sourceLocation: sourceLocation,
+		)
+		return
```
[the remark this answers is not in the document]
Comment (dfed) [reply Tests/SafeDIToolTests/Helpers/SafeDICompileVerifier.swift:63]: Intentional tradeoff. The current behavior records an `Issue` with a pointed message:

> Could not locate SafeDI build artifacts for compilation verification. Ensure the package is built before running tests, or set SAFEDI_SKIP_COMPILE_CHECK=1 to skip.

Silently skipping would let real codegen regressions slip through on filtered runs (e.g. CI shards). Forcing test authors to explicitly opt out via `SAFEDI_SKIP_COMPILE_CHECK` keeps the verifier honest — if someone is running `swift test --filter SafeDIToolTests` in isolation, they get a clear remediation. `swift test` without a filter (the standard path) builds the artifacts as part of the dependency graph, so the guard doesn't fire.

Building on demand from inside the verifier would couple the test harness to the package build system and add non-trivial latency per run. Not worth it when the fix is "build first."

File: Tests/SafeDIToolTests/Helpers/SafeDIToolTestExecution.swift [hunk 20]
```diff
@@ -179,6 +180,15 @@ func executeSafeDIToolTest(
 			nil
 		}
 
+		if let generatedFiles, !generatedFiles.isEmpty, !skipCompileVerification {
+			try verifyGeneratedCodeCompiles(
+				inputSwiftFiles: swiftFiles,
+				additionalDirectorySwiftFiles: additionalDirectoryFiles,
+				generatedFiles: generatedFiles,
```
[the remark this answers is not in the document]
Comment (dfed) [reply Tests/SafeDIToolTests/Helpers/SafeDIToolTestExecution.swift:187]: Valid edge case, but not reachable in the current fixture set. The only test using `includeFolders` is `SafeDIToolCodeGenerationTests` at [line 2949](https://github.com/dfed/SafeDI/blob/claude/compile-verify-codegen-tests/Tests/SafeDIToolTests/SafeDIToolCodeGenerationTests.swift#L2949), which passes `skipCompileVerification: true` because it also uses a placeholder `"Test"` module that doesn't resolve.

If a future fixture needs `--include` directories and compile verification, extending `verifyGeneratedCodeCompiles` to accept a third source list is the right fix — but adding it preemptively with no caller is speculative. Tracking as a latent follow-up.

File: Tests/SafeDIToolTests/Helpers/SafeDIToolTestExecution.swift [hunk 21]
```diff
@@ -179,6 +180,15 @@ func executeSafeDIToolTest(
 			nil
 		}
 
+		if let generatedFiles, !generatedFiles.isEmpty, !skipCompileVerification {
```
[the remark this answers is not in the document]
Comment (dfed) [reply Tests/SafeDIToolTests/Helpers/SafeDIToolTestExecution.swift:183]: Fair point — the input-only pass could still catch fixture-only regressions even when codegen is empty. Keeping as-is for this PR (the verifier's primary job is catching codegen regressions, and every no-output fixture in the suite is either `@Instantiable` without `#SafeDIConfiguration` or a dependency-scan-only test, where input typechecking adds marginal value). Worth revisiting as a follow-up if empty-output fixtures grow in number.

File: Tests/SafeDIToolTests/SafeDIToolMockGenerationErrorTests.swift [hunk 22]
```diff
@@ -348,7 +348,7 @@ struct SafeDIToolMockGenerationErrorTests: ~Copyable {
 				}
 				""",
 				"""
-				@Instantiable(generateMock: true)
+				@Instantiable(generateMock: true, customMock: "customMock")
```
[the remark this answers is not in the document]
Comment (dfed) [reply Tests/SafeDIToolTests/SafeDIToolMockGenerationErrorTests.swift]:
```suggestion
				@Instantiable(generateMock: true, customMockName: "customMock")
```

File: Sources/SafeDICore/Generators/ScopeGenerator.swift [hunk 23]
```diff
@@ -1785,7 +1903,27 @@ actor ScopeGenerator: CustomStringConvertible, Sendable {
...
+					argumentNodePath = nodePath
+				} else if isCycleNode {
+					builderExpression = defaultBuilder
+					optionalBuilderPath = nil
+					argumentNodePath = nodePath
+				} else if !thisOverrideReachable {
+					// Pruned override, no labeled defaults to elide — the full
+					// default builder and the (non-elided) argument list line up.
 					builderExpression = defaultBuilder
 					optionalBuilderPath = nil
```
[the remark this answers is not in the document]
Comment (dfed) [reply Sources/SafeDICore/Generators/ScopeGenerator.swift]: Good catch — fixed on #259 (`4320a97`), which is the home for the back-edge pruning logic. The erased-path emission now branches on `thisOverrideReachable`:

- Unreachable: emit direct construction `\(wrapperType)(\(defaultBuilderCall(arguments:)))` — no optional-binding dance.
- Reachable (but `optionalBuilderPath == nil`, i.e. sendable context where the extracted local IS optional): keep the existing `if let` / `?.()` behavior.

This PR (#256) just enables the compile verifier; the cycle/pruned-back-edge generator work lives on #259.

File: Tests/SafeDIToolTests/SafeDIToolMockGenerationErrorTests.swift [hunk 24]
```diff
@@ -348,7 +348,7 @@ struct SafeDIToolMockGenerationErrorTests: ~Copyable {
 				}
 				""",
 				"""
-				@Instantiable(generateMock: true)
+				@Instantiable(generateMock: true, customMockName: "customMock")
 				public struct Child: Instantiable {
```
[the remark this answers is not in the document]
Comment (dfed) [reply Tests/SafeDIToolTests/SafeDIToolMockGenerationErrorTests.swift]: Stale — that test (`mock_referencesExpectedMockSignature_whenCustomMockMethodMissingDependencyParameters`) was removed from #256 in commit `e49f2ad "remove test testing behavior that is unreachable"`. The macro-rejected Child shape it was pinning can't occur in practice. Current 256 doesn't have this test or any `Child.mock` assertion in the error-tests file.

File: Sources/SafeDICore/Generators/ScopeGenerator.swift [hunk 25]
```diff
@@ -1135,6 +1140,59 @@ actor ScopeGenerator: CustomStringConvertible, Sendable {
...
+				.joined(separator: ", ")
+			let callExpression = if methodName == "init" {
+				"\(concreteType.asSource)(\(labeledArguments))"
+			} else {
+				"\(concreteType.asSource).\(methodName)(\(labeledArguments))"
+			}
+			if typeDescription.propertyType.isSendable {
+				// `@Sendable` attribute forces the closure to satisfy
+				// `@Sendable` captures in parent functions. The `in` keyword
+				// separates the attribute from the body.
```
[the remark this answers is not in the document]
Comment (dfed) [reply Sources/SafeDICore/Generators/ScopeGenerator.swift]: Stale — `defaultBuilderExpressionForPrunedOverride` isn't on this PR. The cycle/pruned-back-edge generator work lives on #259. #256 just wires up the compile verifier; its branch no longer carries any generator changes.

If this concern still applies to the actual implementation on #259, it'd be worth re-flagging there. For context on why you see it here: #256 briefly had #259's commits duplicated on top for testing, and those commits have since been removed from #256.
